analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -0,0 +1,133 @@
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.ParameterReference)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"var b = boolParameter; Tag(""b"", b);", setter).Validator;
+            collector.ValidateTag("b", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToLocalVariable_FromTrackedSymbol_VB()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.ParameterReference)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateVB(@"Dim B As Boolean = BoolParameter : Tag(""B"", B)", setter).Validator;
+            collector.ValidateTag("B", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [DataTestMethod]
+        [DataRow(@"Sample.StaticField = 42; Tag(""Target"", Sample.StaticField);")]
+        [DataRow(@"StaticField = 42; Tag(""Target"", StaticField);")]
+        [DataRow(@"Sample.StaticProperty = 42; Tag(""Target"", Sample.StaticProperty);")]
+        [DataRow(@"StaticProperty = 42; Tag(""Target"", StaticProperty);")]
+        [DataRow(@"var arr = new byte[] { 13 }; arr[0] = 42; Tag(""Target"", arr[0]);")]
+        [DataRow(@"var dict = new Dictionary<string, int>(); dict[""key""] = 42; Tag(""Target"", dict[""key""]);")]
+        [DataRow(@"var other = new Sample(); other.Property = 42; Tag(""Target"", other.Property);")]
+        [DataRow(@"this.Property = 42; Tag(""Target"", this.Property);")]
+        [DataRow(@"Property = 42; Tag(""Target"", Property);")]
+        [DataRow(@"this.field = 42; Tag(""Target"", this.field);")]
+        [DataRow(@"field = 42; Tag(""Target"", this.field);")]
+        public void SimpleAssignment_ToUnsupported_FromLiteral(string snipet)
+        {
+            var collector = SETestContext.CreateCS(snipet, new LiteralDummyTestCheck()).Validator;
+            collector.Validate("Literal: 42", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue("it's scaffolded"));
+            collector.ValidateTag("Target", CheckSymbolicValue);
+
+            static void CheckSymbolicValue(SymbolicValue value)
+            {
+                if (value == null)
+                {
+                    value.Should().BeNull();
+                }
+                else
+                {
+                    value.HasConstraint(DummyConstraint.Dummy).Should().BeFalse();
+                }
+            }
```

Comment:
It doesn't make sense to check if value is null and then assert it is. That assertion can never fail


```suggestion
            collector.ValidateTag("Target", x =>  (x?.HasConstraint(DummyConstraint.Dummy) ?? false).Should().BeFalse());
```